code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm

```diff
@@ -0,0 +1,188 @@
+/mob/living/carbon/xenomorph/hivemind
+	caste_base_type = /mob/living/carbon/xenomorph/hivemind
+	name = "Hivemind"
+	real_name = "Hivemind"
+	desc = "A glorious singular entity."
+
+	icon_state = "hivemind_marker"
+	icon = 'icons/mob/cameramob.dmi'
+
+	status_flags = GODMODE | INCORPOREAL
+	density = FALSE
+	throwpass = TRUE
+
+	a_intent = INTENT_HELP
+
+	health = 1000
+	maxHealth = 1000
+	plasma_stored = 5
+	tier = XENO_TIER_ZERO
+	upgrade = XENO_UPGRADE_ZERO
+
+	mouse_opacity = MOUSE_OPACITY_OPAQUE
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
+	see_invisible = SEE_INVISIBLE_LIVING
+	invisibility = INVISIBILITY_OBSERVER
+	sight = SEE_MOBS|SEE_TURFS|SEE_OBJS
+	see_in_dark = 8
+	move_on_shuttle = TRUE
+
+	hud_type = /datum/hud/hivemind
+	hud_possible = list(PLASMA_HUD, PHEROMONE_HUD, QUEEN_OVERWATCH_HUD)
+
+	var/mob/living/carbon/xenomorph/observed_xeno
+
+/mob/living/carbon/xenomorph/hivemind/Initialize(mapload, ...)
+	. = ..()
+	GLOB.hivemind_list += src
+
+	// Remove some unused xeno things
+	verbs -= /mob/living/proc/lay_down
+
+/mob/living/carbon/xenomorph/hivemind/Destroy()
+	GLOB.hivemind_list -= src
+	return ..()
+
+/mob/living/carbon/xenomorph/hivemind/Move(NewLoc, Dir = 0)
+	var/obj/effect/alien/weeds/W = locate() in range("3x3", NewLoc)
+	if(!W)
+		var/obj/effect/alien/weeds/nearby = locate() in range("3x3", loc)
+		if(!nearby)
+			// If we run out of weeds just teleport to some random weeds.
+			forceMove(get_turf(locate(/obj/effect/alien/weeds)))
```

Comment:
The ideal method to implement this is for weeds and fire to modify flags on the turf they spawn in, and to check for those flags on the turf, I believe.

Reply:
> The ideal method to implement this is for weeds and fire to modify flags on the turf they spawn in, and to check for those flags on the turf, I believe

I dont mind this, do you mind doing a sample of how you'd like turf flags to be like?

Reply:
You don't have to do it, but I'd add a `var/turf_flags = NONE` variable to the base of turfs, and then do some wrapper procs to add and remove the turf traits. Something like this:
```
#define TURF_WEEDED (1<<0)

/turf/proc/set_weeds(weed_state)
  if(weed_state)
    if((turf_flags & TURF_WEEDED))
      return
    turf_flags |= TURF_WEEDED
    //room for changes when weeding happen
    return
  if(!(turf_flags & TURF_WEEDED))
    return
  turf_flags &= ~TURF_WEEDED
  //room for changes when de-weeding happens
```
The same logic would apply to fires and other turf phenomena.
What do you think, @ninjanomnom ?

Reply:
This is one of the harder problems to solve in a good way, what you're suggesting is a form of spaghetti code imo but there aren't many better options if you want this exact behavior.

If you're willing to make the graphic for it though you can tweak how the mechanic works to make the code much cleaner. Just prevent going too far away from the last touched weed and show some tether on the screen. This way you only need to check distance to a single weed every move rather than either track every legal turf or range() check every move.

Reply:
I'm thinking on it a bit broader. We don't have atmos fire, but we have fire and weed objects, which are found with `locate() in turf` each cycle. Wouldn't it make more sense to treat it as a turf property, if it's on fire or weeded?

Reply:
Fire is a bit more broad but something weeded is just a turf with a weed on it, that's not got anything to do with the turf really and is spaghetti due to it being code unrelated to the turf.